resources/install.rb

```diff
@@ -20,15 +20,54 @@
 include IISCookbook::Helper
 
 property :source, String
-property :additional_components, Array, default: []
+property :additional_components, [Array, String],
+ coerce: proc { |c| Array(c) },
+ default: []
+property :install_method, [String, Symbol],
+  required: false,
+  coerce: proc { |i| i.to_sym },
+  equal_to: [:windows_feature_dism, :windows_feature_powershell, :windows_feature_servermanagercmd],
+  default: :windows_feature_dism
+property :start_iis, [true, false], default: false
 
 action :install do
-  features_to_install = ['IIS-WebServerRole'].concat new_resource.additional_components
+  features = ['IIS-WebServerRole'].concat(new_resource.additional_components)
+
+  features_to_install = if new_resource.install_method == :windows_feature_powershell
+                          powershell_feature_name(features)
+                        else
+                          features
+                        end
 
   windows_feature 'Install IIS and additional components' do
     feature_name features_to_install
     action :install
     all !IISCookbook::Helper.older_than_windows2012?
     source new_resource.source unless new_resource.source.nil?
+    install_method new_resource.install_method
+  end
+
+  service 'iis' do
+    service_name 'W3SVC'
+    action [:enable, :start]
+    only_if { new_resource.start_iis }
+  end
+end
+
+action_class do
+  def powershell_feature_name(names)
+    Array(names).map do |name|
+      # This will search for the powershell format (Name) of the feature name, by the both the install name or Name, meaning
+      # that it doesnt care if you pass the powershell format or dism format, it will return the powershell format
+      cmd = "Get-WindowsFeature | Where-Object {$_.AdditionalInfo.InstallName -Eq '#{name}' -or $_.Name -eq '#{name}'} | Select -Expand Name"
+      result = powershell_out cmd
+      if result.stderr.to_s.empty?
+        next result.stdout.strip
+      else
+        Chef::Log.warn("Unable to translate feature #{name}")
```

Comment:
Should this be a silent failure? I think if given an incorrect name it should throw an error else people who do not read the logs may have a different expectation compared to reality